compiler/passes/src/static_single_assignment/rename_expression.rs

```diff
@@ -191,10 +191,10 @@ impl ExpressionConsumer for StaticSingleAssigner<'_> {
         if struct_definition.is_record {
             // Add the `owner` field.
             // Note that the `unwrap` is safe, since type checking guarantees that the member exists.
-            reordered_members.push(member_map.remove(&sym::owner).unwrap());
+            reordered_members.push(member_map.shift_remove(&sym::owner).unwrap());
```

Comment:
You may not need `shift_remove` here since the removals are done for each member in the struct definition.